cmd/kubelet/app/server_test.go

```diff
@@ -61,3 +62,22 @@ func TestValueOfAllocatableResources(t *testing.T) {
 		}
 	}
 }
+
+func TestWriteConfigTo(t *testing.T) {
+	if err := os.MkdirAll("testdata", 0755); err != nil {
+		t.Fatalf("failed to create temporary testdata directory: %v", err)
+	}
+	defer os.Remove("testdata/kubelet.conf")
+	cmd := NewKubeletCommand()
+	if err := cmd.RunE(cmd, []string{"--write-config-to", "testdata/kubelet.conf", "--port", "13131"}); err != nil {
+		t.Fatalf("failed to run kubelet: %v", err)
+	}
+
+	kubeletConfig, err := loadConfigFile("testdata/kubelet.conf")
+	if err != nil {
+		t.Fatalf("failed to load kubelet.conf: %v", err)
+	}
+	if kubeletConfig.Port != 13131 {
+		t.Fatalf("expected port to be %q, but it was %q instead", 13131, kubeletConfig.Port)
+	}
+}
```

Comment:
test case:
1. when  the --write-config-to file not exist.

Reply:
I am not sure I understand the test case you are proposing. Can you please explain? Thanks!

Reply:
example: --write-config-to /path/to/config 
I think you should add test case when /path/to/config not exist.

Reply:
The current test does not expect the config file to exist (in fact, it will overwrite it if it does). I can add a test that fails to write the config file (e.g. because intermediate directories do not exist), but I don't think this adds much value here.

Unless I am missing something?

Reply:
@neoaggelos ok. you are right.